airbyte-integrations/connectors/source-linnworks/source_linnworks/streams.py

```diff
@@ -0,0 +1,227 @@
+        url = urlparse(response.request.url)
+        qs = dict(parse_qsl(url.query))
+
+        page_size = int(qs.get("entriesPerPage", self.page_size))
+        page_number = int(qs.get("pageNumber", 0))
+
+        data = response.json()
+
+        if response.status_code == requests.codes.ok and len(data) == page_size:
+            return {
+                "entriesPerPage": page_size,
+                "pageNumber": page_number + 1,
+            }
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {
+            "entriesPerPage": self.page_size,
+            "pageNumber": 1,
+            "loadCompositeParents": "true",
+            "loadVariationParents": "true",
+            "dataRequirements": "[0,1,2,3,4,5,6,7,8]",
+            "searchTypes": "[0,1,2]",
+        }
+
+        if next_page_token:
+            params.update(next_page_token)
+
+        return params
+
+
+class IncrementalLinnworksStream(LinnworksStream, ABC):
+    state_checkpoint_interval = 100
+
+    @property
+    def cursor_field(self) -> str:
+        return []
+
+    def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        if not stream_state:
+            stream_state = {}
+
+        return [
+            {
+                self.cursor_field: stream_state.get(self.cursor_field, self.start_date),
+            }
+        ]
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        current = current_stream_state.get(self.cursor_field, "")
+        latest = latest_record.get(self.cursor_field, "")
+
+        return {
+            self.cursor_field: max(latest, current),
+        }
+
+
+class ProcessedOrders(LinnworksGenericPagedResult, IncrementalLinnworksStream):
+    # https://apps.linnworks.net/Api/Method/ProcessedOrders-SearchProcessedOrders
+    # Response: SearchProcessedOrdersResponse https://apps.linnworks.net/Api/Class/API_Linnworks-Controllers-ProcessedOrders-Responses-SearchProcessedOrdersResponse
+    # Allows 150 calls per minute
+    primary_key = "nOrderId"
+    cursor_field = "dReceivedDate"
+    page_size = 500
+
+    def path(self, **kwargs) -> str:
+        return "/api/ProcessedOrders/SearchProcessedOrders"
+
+    def request_body_data(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        from_date = stream_slice[self.cursor_field]
+        to_date = pendulum.tomorrow("UTC").isoformat()
+        request = {
+            "DateField": "received",
+            "FromDate": from_date,
+            "ToDate": max(from_date, to_date),
+            "PageNumber": 1,
```

Comment:
```suggestion
            "PageNumber": 1  if not next_page_token else next_page_token["PageNumber"]
```
Not mandatory sugar suggestion 😄